### PR TITLE
Add support for dynamic dropdowns in vector editor

### LINKED ISSUE #8532: No dynamic widgets in table nodes.
image

Both in `aggregate` and `order_by` nodes, the column/aggregates list has 
* default argument when adding new item
* drop-downs inside list
in GUI1, but not in GUI2

Simpler case like `Table.get` also does not display drop down with list of columns.

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

Closes #8532

Adds support for dynamic dropdowns inside the vector editor, providing default value for new items, and also fixing trivial cases with calls like `Table.get`.

https://github.com/enso-org/enso/assets/6566674/583afc02-f476-4345-94ca-41495d1b3e38

### Important Notes

For now, we pass `dynamicConfig` everywhere manually, which requires a lot of boilerplate in each widget. The obvious solution would be to use `provider` to pass dynamic configuration down the widget graph, but I didn’t have time to implement it yet. We can create a separate task for that, shouldn’t be too hard. @Frizi, you asked about my thoughts on this. 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 1]
```diff
@@ -41,15 +44,24 @@ const dynamicTags = computed<Tag[]>(() => {
...
+  if (widgetConfig == null) return []
+  let values: Choice[]
+  if (
+    props.input instanceof VectorItem &&
+    widgetConfig.kind === 'Vector_Editor' &&
+    widgetConfig.item_editor.kind === 'Single_Choice'
+  ) {
+    values = widgetConfig.item_editor.values
+  } else if (widgetConfig.kind === 'Single_Choice') {
+    values = widgetConfig.values
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue:56]: It's very patchy. Now every widget must consider if it happens to be inside vector editor. I think the ListEditor should just pass `item_editor` as dynamicConfig of its elements. Similarly, it's the WidgetFunction/Application which should extract the piece of config for every argument.

File: app/gui2/src/components/GraphEditor/widgets/WidgetVector.vue [hunk 2]
```diff
@@ -6,21 +6,43 @@ import { ForcePort } from '@/providers/portInfo'
...
+
+const defaultConstructor = computed(() => {
+  const fallback = Ast.Wildcard.new
+  const config = props.config
+  if (config == null) return fallback
+  const [_, widgetConfig] = config.find(([name]) => name === argInfo.value?.name) ?? []
+  if (
+    widgetConfig &&
+    widgetConfig.kind == 'Vector_Editor' &&
+    widgetConfig.item_editor.kind === 'Single_Choice'
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor/widgets/WidgetVector.vue:28]: Why the item_editor must be a `Single_Choice`? I thought the default can be just any value, e.g. number for number slider.

**Review by farmaazon — CHANGES_REQUESTED** [review]
I see serious flaws in current implementation. And it's still very buggy:
1. If I add node `order_by` and only after realize I miss the list (as the placeholder does not allow making entries yet) and edit node to add `[]` then no dynamic config seems to work in the node:
image

2. In aggregate, the default value works, but Group_By does not make space for column name

image

3. Picking values in drop-downs inside list editors does not seem to change anything

**Comment by vitvakatu** [comment]
Refactored and included in https://github.com/enso-org/enso/pull/8624

**Pull request closed without merging**